mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreRightManagerTest.java

```diff
@@ -24,17 +24,176 @@
 import static org.apache.james.mailbox.fixture.MailboxFixture.CEDRIC;
 import static org.apache.james.mailbox.fixture.MailboxFixture.INBOX_ALICE;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import javax.mail.Flags;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.james.mailbox.acl.GroupMembershipResolver;
+import org.apache.james.mailbox.acl.MailboxACLResolver;
+import org.apache.james.mailbox.acl.SimpleGroupMembershipResolver;
+import org.apache.james.mailbox.acl.UnionMailboxACLResolver;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.exception.MailboxNotFoundException;
 import org.apache.james.mailbox.exception.UnsupportedRightException;
 import org.apache.james.mailbox.mock.MockMailboxSession;
 import org.apache.james.mailbox.model.MailboxACL;
 import org.apache.james.mailbox.model.MailboxACL.Right;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
 import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 public class StoreRightManagerTest {
 
     public static final long UID_VALIDITY = 3421l;
+    private StoreRightManager storeRightManager;
+    private MailboxACLResolver mailboxAclResolver;
+    private GroupMembershipResolver groupMembershipResolver;
+    private String alice;
+    private MockMailboxSession aliceSession;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    private MailboxMapper mockedMailboxMapper;
+
+    @Before
+    public void setup() throws MailboxException {
+        alice = "Alice";
+        aliceSession = new MockMailboxSession(alice);
+        MailboxSessionMapperFactory mockedMapperFactory = mock(MailboxSessionMapperFactory.class);
+        mockedMailboxMapper = mock(MailboxMapper.class);
+        mailboxAclResolver = new UnionMailboxACLResolver();
+        groupMembershipResolver = new SimpleGroupMembershipResolver();
+        when(mockedMapperFactory.getMailboxMapper(aliceSession))
+            .thenReturn(mockedMailboxMapper);
+        storeRightManager = new StoreRightManager(mockedMapperFactory,
+                                                  mailboxAclResolver,
+                                                  groupMembershipResolver);
+    }
+
+    @Test
+    public void hasRighShouldThrowMailboxNotFoundExceptionWhenMailboxDoesNotExist() throws MailboxException {
```

Comment:
s/hasRigh/hasRight/ everywhere in this file

Reply:
Raphael is the new spell checker ;-)